src/editor/validator.ts

```diff
@@ -22,7 +21,47 @@ export class Validator {
         this.module = module;
     }
 
-    // canBackspaceCurLine()
+    /**
+     * logic: checks if at the end of an editable text
+     * OR selected an empty editable text / identifier
+     * OR right before an editable item and need to select it
+     */
+    canMoveToNextTokenAtTextEditable(providedContext?: Context): boolean {
+        const context = providedContext ? providedContext : this.module.focus.getContext();
+
+        return (
+            ((context.tokenToRight instanceof IdentifierTkn || context.tokenToRight instanceof EditableTextTkn) &&
+                context.tokenToRight.isEmpty) ||
+            context.tokenToLeft instanceof EditableTextTkn ||
+            context.tokenToLeft instanceof IdentifierTkn ||
+            (context.token?.isEmpty && context.selected)
+        );
+    }
+
+    /**
+     * logic: checks if at the beginning of an editable text
+     * OR selected an empty editable text / identifier
+     */
+    canMoveToPrevTokenAtTextEditable(providedContext?: Context): boolean {
+        const context = providedContext ? providedContext : this.module.focus.getContext();
+
+        return (
+            (context.tokenToLeft instanceof IdentifierTkn && context.tokenToRight.isEmpty) ||
```

Comment:
When does this top case happen? When we are here `abc| = 123`?

Reply:
Hmmm, good question. I really don't remember. let me check

Reply:
Ok, I've found out: When we are at `|---` (not selected the empty token, but having the cursor at the beginning of the empty identifier) => this is usually the case when the user wants to either enter a new line at the beginning of a variable declaration statement.

Reply:
I see. Thanks! These are good to know.